src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1173,7 +1173,7 @@ private void Dispose()
                     bool stoppedAtSIGSTOP = false;
                     if (sigName == "SIGSTOP")
                     {
-                        if (AD7Engine.RemoveChildProcess(_launchOptions.ProcessId))
+                        if (AD7Engine.RemoveChildProcess(_launchOptions.ProcessId.HasValue ? _launchOptions.ProcessId.Value : 0))
```

Comment:
>RemoveChildProcess [](start = 38, length = 18)

I think you want to change the check on line 1175 to include `&& ProcessId.HasValue`.